app/views/LoginView/UserForm.tsx

```diff
@@ -0,0 +1,170 @@
+import sharedStyles from '../Styles';
+import UGCRules from '../../containers/UserGeneratedContentRules';
+import { useAppSelector } from '../../lib/hooks';
+import styles from './styles';
+
+interface ISubmit {
+	user: string;
+	password: string;
+}
+
+const schema = yup.object().shape({
+	user: yup.string().required(),
+	password: yup.string().required()
+});
+
+export interface UserFormRef {
+	getUser: () => string;
+}
+
+const UserForm = forwardRef<UserFormRef>((_, ref) => {
+	const passwordInput = useRef<RNTextInput | null>(null);
+
+	const {
+		params: { username }
+	} = useRoute<RouteProp<OutsideParamList, 'LoginView'>>();
+	const navigation = useNavigation<StackNavigationProp<OutsideParamList, 'LoginView'>>();
+
+	const {
+		control,
+		handleSubmit,
+		formState: { isValid },
+		getValues
+	} = useForm<ISubmit>({ mode: 'onChange', resolver: yupResolver(schema), defaultValues: { user: username || '' } });
+
+	const { colors } = useTheme();
+	const dispatch = useDispatch();
+
+	const getUser = () => getValues('user');
+	useImperativeHandle(ref, () => ({
+		getUser
+	}));
+
+	const {
+		Accounts_EmailOrUsernamePlaceholder,
+		Accounts_PasswordPlaceholder,
+		Accounts_PasswordReset,
+		Accounts_RegistrationForm_LinkReplacementText,
+		isFetching,
+		Accounts_RegistrationForm,
+		Site_Name,
+		inviteLinkToken
+	} = useAppSelector(state => ({
+		Accounts_RegistrationForm: state.settings.Accounts_RegistrationForm as string,
+		Accounts_RegistrationForm_LinkReplacementText: state.settings.Accounts_RegistrationForm_LinkReplacementText as string,
+		isFetching: state.login.isFetching,
+		Accounts_EmailOrUsernamePlaceholder: state.settings.Accounts_EmailOrUsernamePlaceholder as string,
+		Accounts_PasswordPlaceholder: state.settings.Accounts_PasswordPlaceholder as string,
+		Accounts_PasswordReset: state.settings.Accounts_PasswordReset as boolean,
+		Site_Name: state.settings.Site_Name as string,
+		inviteLinkToken: state.inviteLinks.token
+	}));
+
+	const showRegistrationButton =
+		Accounts_RegistrationForm === 'Public' || (Accounts_RegistrationForm === 'Secret URL' && inviteLinkToken?.length);
+
+	const register = () => {
+		navigation.navigate('RegisterView', { title: Site_Name });
+	};
+
+	const forgotPassword = () => {
+		navigation.navigate('ForgotPasswordView', { title: Site_Name });
+	};
+
+	const submit = ({ password, user }: ISubmit) => {
+		if (!isValid) {
+			return;
+		}
+		Keyboard.dismiss();
+		dispatch(loginRequest({ user, password }));
+	};
+
+	return (
+		<>
+			<Text style={[styles.title, sharedStyles.textBold, { color: colors.titleText }]}>{I18n.t('Login')}</Text>
+			<ControlledFormTextInput
+				name='user'
+				control={control}
+				label={I18n.t('Username_or_email')}
+				containerStyle={styles.inputContainer}
+				placeholder={Accounts_EmailOrUsernamePlaceholder || I18n.t('Username_or_email')}
+				keyboardType='email-address'
+				returnKeyType='next'
+				onSubmitEditing={() => {
```

Comment:
```suggestion
				onSubmitEditing={() => passwordInput.current?.focus()}
```
Not to change, but just a comment, this approach reduces the number of lines of code, making it easier to read.